src/main/java/com/metamx/metrics/SigarUtil.java

```diff
@@ -58,9 +58,20 @@
     }
   }
 
+  /**
+   * SigarHolder class is initialized after SigarUtil, that guarantees that SIGAR = new Sigar() is executed after static
+   * block (which loads the library) of SigarUtil is executed. This is anyway guaranteed by JLS if SIGAR static field
+   * goes below the static block in textual order, but fragile e. g. if someone applies automatic reformatting and the
+   * static field is moved above the static block.
+   */
+  private static class SigarHolder
+  {
+    private static final Sigar SIGAR = new Sigar();
+  }
+
   public static Sigar getSigar()
   {
-    return new Sigar();
+    return SigarHolder.SIGAR;
```

Comment:
apparently Sigar computes diffs of cpu/sys/user itself, and without any synchronization. Cassandra uses it as singleton because it doesn't expose this functionality of Sigar class. We don't use it directly too, but since we openly return Sigar class and this is a "library", it is safer not to try to cache Sigar here, in order to avoid having concurrency bugs in future is someone uses Sigar improperly. Anyway the cost of a single Sigar instance initialization is not so high, and we instantiate just a few such instances per JVM (one per XxxMonitor instance).